### PR TITLE
Add javadoc commands

### PR DESCRIPTION
**Pull request by OroArmor**
This is just the start

### CODE REVIEW

**Review by gdude2002 — CHANGES_REQUESTED** [review]
You're making use of Java networking APIs, which is not appropriate. Instead, you should be making use of KTor - you can find an example of how to create and use a KTor client in the Minecraft extension.

This otherwise seems more or less fine - though your use of `simple-xml` for serialization introduces a reliance on reflection that would be completely avoided by making use of `kotlinx.serialization`, something which you implied was possible via a project you linked on Discord. Using `kotlinx.serialization` should also integrate with KTor.

File: src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/JavadocExtension.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		GUILDS.forEach { guildId ->
+			publicSlashCommand {
+				name = "javadoc"
+				description = "Get Javadocs for common Quilt Projects"
+
+				allowInDms = true
+
+				guild(guildId)
+
+				action {
```
Comment (gdude2002) [inline src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/JavadocExtension.kt:49]: This action block can never run. Discord does not support executable root commands when there are groups or subcommands. I'm pretty sure KordEx throws when you do this - you do need to actually test your code.

**Review by gdude2002 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/JavadocExtension.kt (hunk 1 above)
Comment (OroArmor) [reply src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/JavadocExtension.kt:49]: That's annoying but ok. Ill fix this today

**Comment by OroArmor** [comment]
Is there anyone who is able to test this? The setup to get cozy working is a little complex and I don't need to test everything.

**Comment by OroArmor** [comment]
Alright, this PR has been tested and it works. However, it isn't very polished and I don't know the best way to polish it. For instance, it will always return the most recent version of the project, not the most relevant (ie mappings for 22w44a and not 1.19.2). I could implement the versioning system from Linkie, but I feel like that is growing the scope too much. As for this working, it does.

**Comment by OroArmor** [comment]
I didn't click any of the links, and each time I do it pops up this:
image

This is a problem, and I'm not sure the best way to fix it. Maybe some routing from the main QuiltMC website?

**Comment by gdude2002** [comment]
This has long been inactive, so I'm closing it for now.

I think this needs to be redesigned based on the new Maven system - I'm not sure you can currently browse JDs there, and, well, we'd need new URLs to avoid the popup anyway.

Maybe this should be replaced with an actual JD lookup tool and not just a link retriever.

**Pull request closed without merging**